From a0540ba674d206c744f83d6c2c96b4a58c1ac037 Mon Sep 17 00:00:00 2001
From: Benjamin Berg <bberg@redhat.com>
Date: Thu, 10 Sep 2020 18:55:16 +0200
Subject: [PATCH] adaptive: Log name of unknown condition if known
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Printing a number if not very helpful, and most likely we have a known
name for it, so use that.
---
 src/thd_engine_adaptive.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/thd_engine_adaptive.cpp b/src/thd_engine_adaptive.cpp
index 40a6e7d..991a79f 100644
--- a/src/thd_engine_adaptive.cpp
+++ b/src/thd_engine_adaptive.cpp
@@ -789,6 +789,8 @@ int cthd_engine_adaptive::parse_gddv(char *buf, int size, int *end_offset) {
 }
 
 int cthd_engine_adaptive::verify_condition(struct condition condition) {
+	const char *cond_name;
+
 	if (condition.condition >= Oem0 && condition.condition <= Oem5)
 		return 0;
 	if (condition.condition >= adaptive_condition(0x1000)
@@ -810,7 +812,8 @@ int cthd_engine_adaptive::verify_condition(struct condition condition) {
 	if (condition.condition == Platform_type)
 		return 0;
 
-	thd_log_error("Unsupported condition %d\n", condition.condition);
+	cond_name = condition_names[MIN(MAX(0, condition.condition), G_N_ELEMENTS(condition_names) - 1)];
+	thd_log_error("Unsupported condition %d (%s)\n", condition.condition, cond_name);
 	return THD_ERROR;
 }
 
-- 
2.27.0

